engines/bops_core/spec/controllers/bops_core/auditable_controller_spec.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require "bops_core_helper"
+
+RSpec.describe BopsCore::AuditableController, type: :controller do
+  controller(ActionController::Base) do
+    include BopsCore::AuditableController
+
+    audit :create, event: "event", scope: "scope", payload: -> { {foo: "bar"} }
+
+    def create
+    end
+  end
+
+  it "sends an audit event for the create action" do
+    expect do |block|
+      ActiveSupport::Notifications.subscribe("event.scope.bops_audit", &block)
+
+      post :create
+    end.to yield_with_args(
+      "event.scope.bops_audit",
+      an_instance_of(Time),
+      an_instance_of(Time),
+      a_string_matching(/^[a-z0-9]{20}$/),
+      a_hash_including(foo: "bar")
+    )
+  end
```

Comment:
Looking to create a matcher that wraps all this up - the arity of the block passed to subscribe affects the behaviour and in this case the RSpec `yield_with_args` matcher has a `*args` arity so the event object gets splatted to individual args.